src/MonolithicNavierStokes.cpp

```diff
@@ -0,0 +1,631 @@
+                    // ------
+                    cell_system_matrix(i, j) += fe_values[pressure].value(i, q) * fe_values[velocity].divergence(j, q) * fe_values.JxW(q);
+
+                    // Mass matrix for the pressure
+                    // ------
+                    // Mp_ij = ∫ (1/ν)ψ_i ψ_j dx
+                    // ------
+                    pressure_mass_cell_matrix(i, j) += fe_values[pressure].value(j, q) * fe_values[pressure].value(i, q) / nu * fe_values.JxW(q);
+
+                    // Mass matrix for the velocity
+                    // ------
+                    // Mv_ij = ∫ (1/Δt) φ_i·φ_j dx
+                    // ------
+                    velocity_mass_cell_matrix(i, j) += scalar_product(fe_values[velocity].value(i, q), fe_values[velocity].value(j, q)) / deltat * fe_values.JxW(q);
+                }
+            }
+        }
+        cell->get_dof_indices(dof_indices);
+
+        system_matrix.add(dof_indices, cell_system_matrix);
+        velocity_mass.add(dof_indices, velocity_mass_cell_matrix);
+        pressure_mass.add(dof_indices, pressure_mass_cell_matrix);
+    }
+    system_matrix.compress(VectorOperation::add);
+    pressure_mass.compress(VectorOperation::add);
+    velocity_mass.compress(VectorOperation::add);
+}
+
+template <unsigned int dim>
+void MonolithicNavierStokes<dim>::add_convective_term()
+{
+    const unsigned int dofs_per_cell = fe->dofs_per_cell;
+    const unsigned int n_q = quadrature->size();
+
+    FEValues<dim> fe_values(*fe,
+                            *quadrature,
+                            update_values | update_gradients |
+                                update_quadrature_points | update_JxW_values);
+
+    FullMatrix<double> cell_lhs_matrix(dofs_per_cell, dofs_per_cell);
+
+    std::vector<types::global_dof_index> dof_indices(dofs_per_cell);
+
+    std::vector<Tensor<1, dim>> previous_velocity_values(n_q);
+    std::vector<double> previous_velocity_divergence(n_q);
+
+    lhs_matrix = 0.0;
+
+    lhs_matrix.copy_from(system_matrix);
+
+    for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+        if (!cell->is_locally_owned())
+            continue;
+
+        fe_values.reinit(cell);
+
+        cell_lhs_matrix = 0.0;
+
+        fe_values[velocity].get_function_values(solution, previous_velocity_values);
+        fe_values[velocity].get_function_divergences(solution, previous_velocity_divergence);
+
+        for (unsigned int q = 0; q < n_q; ++q)
+        {
+            for (unsigned int i = 0; i < dofs_per_cell; ++i)
+            {
+                for (unsigned int j = 0; j < dofs_per_cell; ++j)
+                {
+                    // Non-linear term
+                    // ------
+                    // N_ij(u*) = ∫ (u*·∇u)·φ_j dx
+                    // ------
+                    cell_lhs_matrix(i, j) += scalar_product(fe_values[velocity].gradient(j, q) * previous_velocity_values[q], fe_values[velocity].value(i, q)) * fe_values.JxW(q);
```

Comment:
Long lines such as this one are hard to read and to debug. I encourage you to install automatic formatting tools such as clang-format in your favorite text editor.